src/DependencyInjection/SonataMediaExtension.php

```diff
@@ -528,62 +481,10 @@ public function configureExtra(ContainerBuilder $container, array $config)
     }
 
     /**
-     * Add class to compile.
+     * NEXT_MAJOR: remove this method.
```

Comment:
We should also trigger explicit deprecation messages (`E_USER_DEPRECATED`).

Reply:
I guess `@deprecated` is enough

Reply:
I'm not sure about that, because the conversion of this annotation in a real error depends on the usage of the [Debug component](https://symfony.com/doc/current/components/debug.html#debugging-a-class-loader) (`DebugClassLoader`). So, if the users are not using this component, they will never be notified about the deprecation.

Reply:
I followed [the contributing doc](https://github.com/sonata-project/dev-kit/blob/master/project/CONTRIBUTING.md#deprecation), if so I guess it should be changed as well

Reply:
I think it's fine to assume this component is in use.

Reply:
In that case, we should check if we can elaborate a rule for these kind of scenarios where we are deprecating a method or a class. Maybe removing the `trigger_error()` calls.